lib/seahorn/BvOpSem2TrackingRawMemMgr.hh

```diff
@@ -66,19 +101,36 @@ public:
 
     RawMemValTy getRaw() { return strct::extractVal(m_v, 0); }
 
-    Expr getMetadata() { return strct::extractVal(m_v, 1); }
+    Expr getMetadata(MetadataKind kind) {
+      return strct::extractVal(m_v, kind + 1);
```

Comment:
fragile. order of metadata and order of enum can be different. at least comment somewhere that they should match. But better, add a function that converts kind to index in tracking memory manager and use that function.

Reply:
Added a comment for now. This can be made less fragile but I want to solve it at the right level. 

e.g. Ideally there should be one class that deals with arranging metadata memory and all client  - MemMgr, MemToExpr classes should use its method rather than knowing internal details. We are not there yet and changing it as part of this feature appears out of scope to me.